wasmbinding/query_plugin.go

```diff
@@ -138,6 +173,32 @@ func CustomQuerier(qp *QueryPlugin) func(ctx sdk.Context, request json.RawMessag
 	}
 }
 
+// NormalizeReponses normalizes the responses by unmarshalling the response then marshalling them again.
+// Normalizing the response is specifically important for responses that contain type of Any.
+func NormalizeResponseAndJsonMarshal(binding interface{}, bz []byte, codec codec.Codec) ([]byte, error) {
+	// all values are proto message
+	message, ok := binding.(proto.Message)
+	if !ok {
+		return nil, wasmvmtypes.Unknown{}
+	}
+
+	// unmarshal binary into stargate response data structure
+	err := proto.Unmarshal(bz, message)
+	if err != nil {
+		return nil, wasmvmtypes.Unknown{}
+	}
+
+	bz, err = codec.MarshalJSON(message)
```

Comment:
We json marshal the response here for the contracts because stargate querier json unmarshals the response by default

Reply:
Just to see if I understand this correctly: this method converts the encoding from protobuf (received from the `route` call above) to json, which is what wasmd expects. Is that correct? If so, I'd like to suggest renaming this method from `normalize` to something like `protoToJson`

Reply:
makes sense! Thanks for the recommendation :) fixed!